<!DOCTYPE HTML>
<html>
<head>
  <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  
  <title>Buttonwood</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
  
    <meta name="author" content="Buttonwood">
  
  
  <meta name="description" content="Buttonwood | Data Mining | python | perl | Bioinformatics">
<meta property="og:type" content="website">
<meta property="og:title" content="Buttonwood">
<meta property="og:url" content="http://buttonwood.github.io/page/3/index.html">
<meta property="og:site_name" content="Buttonwood">
<meta property="og:description" content="Buttonwood | Data Mining | python | perl | Bioinformatics">
<meta property="og:locale" content="default">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Buttonwood">
<meta name="twitter:description" content="Buttonwood | Data Mining | python | perl | Bioinformatics">
  
    <link rel="alternate" href="/atom.xml" title="Buttonwood" type="application/atom+xml">
  
  
    <link rel="icon" type="image/x-icon" href="/favicon.ico">
  
  <link rel="stylesheet" href="/css/style.css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]--><!-- hexo-inject:begin --><!-- hexo-inject:end -->
  
</head>

<body>
  <!-- hexo-inject:begin --><!-- hexo-inject:end --><div class="wrapper">
    <header id="header">
  <div class="title">
    <h1><a href="/">Buttonwood</a></h1>
    <p><a href="/">Being A Coding Poet. Less Talk, and More Move!</a></p>
  </div>
  <nav class="nav">
    <ul>
      
        <li><a href="/about/">About</a></li>
      
        <li><a href="/archives/">Archives</a></li>
      
        <li><a href="/links/">Links</a></li>
      
      
        <li><a href="/atom.xml">RSS</a></li>
      
    </ul>
    <div class="clearfix"></div>
  </nav>
  <div class="clearfix"></div>
</header>
    <div class="content">




  
    <article class="post">
  <header>
    
      <div class="icon"></div>
      <a href="/2013/06/22/2013-06-22-SSPACE-LongRead/">
  <time datetime="2013-06-22T03:51:00.000Z">
    Jun 22 2013
  </time>
</a>
    
    
  
    <h1 class="title"><a href="/2013/06/22/2013-06-22-SSPACE-LongRead/">SSPACE-LongRead: scaffolding with long reads</a></h1>
  

  </header>
  
  <div class="entry">
    
      <h4 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h4><p>We are happy to say that SSPACE is ready for dealing with the PacBio long reads for scaffolding<sup><a href="#fn_1" id="reffn_1">1</a></sup>.</p>
<p>They proposed a novel hybrid assembly methodology that aims to scaffold pre-assembled contigs in an iterative manner using PacBio RS long read information as a backbone. </p>
<p>The SSPACE-LongRead software which is designed to upgrade incomplete draft genomes using single molecule sequences. We conclude that the recent advances of the PacBio sequencing technology and chemistry, in combination with the limited computational resources required to run our program, allow to scaffold genomes in a fast and reliable manner.</p>
    
  </div>
  <footer class="end-sep">
    
      
        <div class="alignleft">
          <a href="/2013/06/22/2013-06-22-SSPACE-LongRead/#more" class="more-link">Read More</a>
        </div>
      
      
    
    <div class="clearfix"></div>
  </footer>
</article>



  
    <article class="post">
  <header>
    
      <div class="icon"></div>
      <a href="/2013/05/17/2013-05-17-circos/">
  <time datetime="2013-05-16T16:00:00.000Z">
    May 17 2013
  </time>
</a>
    
    
  
    <h1 class="title"><a href="/2013/05/17/2013-05-17-circos/">Circos for Comparative genomics</a></h1>
  

  </header>
  
  <div class="entry">
    
      <h3 id="Synteny-and-Comparative-genomics"><a href="#Synteny-and-Comparative-genomics" class="headerlink" title="Synteny and Comparative genomics"></a>Synteny and Comparative genomics</h3><p>Welcome to see details in my <a href="https://github.com/Buttonwood/Bioinformatics/issues/4" target="_blank" rel="noopener">github issues</a>.</p>
<p>First, use <a href="http://www.bx.psu.edu/miller_lab/dist/README.lastz-1.02.00/README.lastz-1.02.00a.html" target="_blank" rel="noopener">Lastz</a> for synteny block alignment.</p>
<p>Then use SVG for synteny block drawing.<br><img src="https://dl.dropboxusercontent.com/u/78977353/hexo-git/cg/line.png" alt="line"></p>
<p>And we found <a href="http://circos.ca/" target="_blank" rel="noopener">Circos</a> more powerful on handing this.</p>
<p><img src="https://dl.dropboxusercontent.com/u/78977353/hexo-git/cg/circos.png" alt="circos"></p>
<p>说明,本文以下内容多数来自网络资源转载,像原作者致意,感谢他们的原创工作!如想详细了解，请点击衔接，接入原文！</p>
<h3 id="Circos-Installation"><a href="#Circos-Installation" class="headerlink" title="Circos Installation"></a>Circos Installation</h3><p>For OSX: you can refer to this <a href="http://kylase.github.io/CircosAPI/os-x-installation-guide/" target="_blank" rel="noopener">os-x-installation-guide</a> and <a href="http://pgfe.umassmed.edu/ou/archives/2513" target="_blank" rel="noopener">this</a>.</p>
<h3 id="Read-this-first"><a href="#Read-this-first" class="headerlink" title="Read this first"></a>Read this first</h3><p><a href="http://pgfe.umassmed.edu/ou/archives/2513" target="_blank" rel="noopener">Circos系列教程一安装</a></p>
<p><a href="http://pgfe.umassmed.edu/ou/archives/2519" target="_blank" rel="noopener">Circos系列教程二染色体示意图ideograms</a></p>
<p><a href="http://pgfe.umassmed.edu/ou/archives/2554" target="_blank" rel="noopener">Circos系列教程三突出标记Highlight</a></p>
<p><a href="http://pgfe.umassmed.edu/ou/archives/2579" target="_blank" rel="noopener">Circos系列教程四连线links</a></p>
<h3 id="Tutorial"><a href="#Tutorial" class="headerlink" title="Tutorial"></a>Tutorial</h3><p><a href="http://blog.csdn.net/skenoy/article/details/8696402" target="_blank" rel="noopener"> CIRCOS教程翻译 1.1——helloworld</a><br><img src="http://img.my.csdn.net/uploads/201303/20/1363765909_2834.png" alt="1.1"></p>
<p><a href="http://blog.csdn.net/skenoy/article/details/8697823" target="_blank" rel="noopener">CIRCOS教程翻译 1.2——ticks</a><br><img src="http://img.my.csdn.net/uploads/201303/20/1363778829_5544.png" alt="1.2"></p>
<p><a href="http://blog.csdn.net/skenoy/article/details/8716127" target="_blank" rel="noopener">CIRCOS教程翻译 1.3 ——染色体的变化</a><br><img src="http://img.my.csdn.net/uploads/201303/25/1364182311_2969.png" alt="1.3"></p>
<p><a href="http://blog.csdn.net/skenoy/article/details/8716632" target="_blank" rel="noopener">CIRCOS教程翻译 1.4——links和rules</a><br><img src="http://img.my.csdn.net/uploads/201303/25/1364190625_3670.png" alt="1.4"></p>
    
  </div>
  <footer class="end-sep">
    
      
        <div class="alignleft">
          <a href="/2013/05/17/2013-05-17-circos/#more" class="more-link">Read More</a>
        </div>
      
      
    
    <div class="clearfix"></div>
  </footer>
</article>



  
    <article class="post">
  <header>
    
      <div class="icon"></div>
      <a href="/2013/05/02/2013-05-02-de-novo-assembly-of-highly-heterozygous-genomes/">
  <time datetime="2013-05-02T03:51:00.000Z">
    May 2 2013
  </time>
</a>
    
    
  
    <h1 class="title"><a href="/2013/05/02/2013-05-02-de-novo-assembly-of-highly-heterozygous-genomes/"> 杂合基因组的组装（De novo assembly of highly heterozygous genomes )</a></h1>
  

  </header>
  
  <div class="entry">
    
      <h4 id="杂合基因组的组装（De-novo-assembly-of-highly-heterozygous-genomes-）"><a href="#杂合基因组的组装（De-novo-assembly-of-highly-heterozygous-genomes-）" class="headerlink" title="杂合基因组的组装（De novo assembly of highly heterozygous genomes ）"></a>杂合基因组的组装（De novo assembly of highly heterozygous genomes ）</h4><p>最近看到一篇文献<br><a href="http://genome.cshlp.org/content/early/2014/04/22/gr.170720.113?top=1" target="_blank" rel="noopener">Efficient de novo assembly of highly heterozygous genomes from whole-genome shotgun short reads </a>。<br>感觉高杂合的大型基因组的组装有了一些可喜的进展，日本人的工作还是很扎实的，有一些值得借鉴和参考的地方。</p>
<h4 id="Abstract"><a href="#Abstract" class="headerlink" title="Abstract"></a>Abstract</h4><p>随着测序平台的改进和发展，测序通量已经不是问题，测序价格越来越便宜，对于一些非模式生物或野生物种来说，测定它们的基因组序列对于科学研究意义越来越明显。但是，大多数情况来看，它们通常又具有较高的杂合或者多倍性这些问题，这对于以短reads做组装为主流的<strong><em>de novo</em></strong>项目非常棘手，目前尚缺乏较完善兼具可实践的方案(经费多的实验室除外)。</p>
<p>一般来说，杂合基因组的解决有两种可行的方案，但都公认为比较费时费力费钱费脑.</p>
<blockquote>
<ol>
<li>Fosmid-based (or Bac-based) hierarchical sequencing;</li>
<li>Inbred lines ( doubled-monoploid clone).</li>
</ol>
</blockquote>
<p><strong><em>Fosmid或者bac</em></strong>为基础的分层组装需要构建大量的长片段文库，实验工作以及组装拼接都是精细活，像牡蛎<em>oyster</em> (Zhang et al. 2012), 小菜饿<em>diamondback moth</em> (You et al. 2013), and 挪威云杉<em>Norway spruce</em> (Nysted et al. 2013)这些经典案例都值得看一看、读一读。</p>
    
  </div>
  <footer class="end-sep">
    
      
        <div class="alignleft">
          <a href="/2013/05/02/2013-05-02-de-novo-assembly-of-highly-heterozygous-genomes/#more" class="more-link">Read More</a>
        </div>
      
      
    
    <div class="clearfix"></div>
  </footer>
</article>



  
    <article class="post">
  <header>
    
      <div class="icon"></div>
      <a href="/2013/03/20/2013-03-20-install-matplotlib-is-a-nightmare/">
  <time datetime="2013-03-20T14:51:00.000Z">
    Mar 20 2013
  </time>
</a>
    
    
  
    <h1 class="title"><a href="/2013/03/20/2013-03-20-install-matplotlib-is-a-nightmare/">Install matplotlib is a nightmare</a></h1>
  

  </header>
  
  <div class="entry">
    
      <p>Building <strong><em>matplotlib</em></strong> on <strong><em>OSX</em></strong> has proved to be a nightmare because of the different types of zlib, png and freetype that may be on your system.            </p>
<p>The recommended and supported way to build is to use a third-party<br>package manager to install the required dependencies, and then<br>install matplotlib from source using the setup.py script. Two widely<br>used package managers are homebrew and MacPorts. The following<br>example illustrates how to install libpng and freetype using<br>homebrew.</p>
<p>Example usage::     </p>
<pre><code>brew install libpng freetype
</code></pre><p>If you are using MacPorts, execute the following instead:</p>
<p>Example usage::</p>
<pre><code>port install libpng freetype
</code></pre><p>To install matplotlib from source, execute:</p>
<p>Example usage::</p>
<pre><code>  python setup.py install
</code></pre><p>结果freetype报错:</p>
<pre><code>/usr/local/include/ft2build.h:56:10: fatal error: ‘freetype/config/ftheader.h’ file not found
</code></pre><p>google 之在<a href="http://stackoverflow.com/questions/6800488/installing-matplotlib-in-mac-osx-lion" target="_blank" rel="noopener">stackoverflow</a><br>仍然不成功；</p>
<p>最后到处试试，终于在这篇博客看到；<br><a href="http://blog.caoyuan.me/2012/08/matplotlib-error-mac-os-x/" target="_blank" rel="noopener">http://blog.caoyuan.me/2012/08/matplotlib-error-mac-os-x/
</a></p>
<p>最后终于成功！</p>

    
  </div>
  <footer class="end-sep">
    
      
      
    
    <div class="clearfix"></div>
  </footer>
</article>



  
    <article class="post">
  <header>
    
      <div class="icon"></div>
      <a href="/2013/03/06/2013-03-06-i-am-coming-back/">
  <time datetime="2013-03-06T14:21:00.000Z">
    Mar 6 2013
  </time>
</a>
    
    
  
    <h1 class="title"><a href="/2013/03/06/2013-03-06-i-am-coming-back/">I am coming back</a></h1>
  

  </header>
  
  <div class="entry">
    
      <p><strong>Cuz</strong> I was busy these days, I spent a lot of time for updating these beautiful theme<br>for my blog. I like it very much!<br>And here I am testing some code insertion.<br>
    
  </div>
  <footer class="end-sep">
    
      
        <div class="alignleft">
          <a href="/2013/03/06/2013-03-06-i-am-coming-back/#more" class="more-link">Read More</a>
        </div>
      
      
    
    <div class="clearfix"></div>
  </footer>
</article>



  
    <article class="post">
  <header>
    
      <div class="icon"></div>
      <a href="/2013/01/10/2013-01-10-assemble-a-genome/">
  <time datetime="2013-01-10T02:04:00.000Z">
    Jan 10 2013
  </time>
</a>
    
    
  
    <h1 class="title"><a href="/2013/01/10/2013-01-10-assemble-a-genome/">Assemble a genome</a></h1>
  

  </header>
  
  <div class="entry">
    
      <p>基因组拼接就是将测序得到的短 <strong>reads</strong> 还原成更长基因组序列的过程，<br>不同组装软件和组装策略采用的具体算法和细节不尽相同，<br>但总体上都经过如下几步：</p>
<h2 id="a-Contig-组装"><a href="#a-Contig-组装" class="headerlink" title="a) Contig 组装"></a><strong>a) Contig 组装</strong></h2><p>首先，利用 <strong>reads</strong> 间 <strong>overlap</strong> 和覆盖度情况，拼接出 <strong>contigs</strong> 序列；<br><strong>Contigs</strong> 组装方法较多，软件丰富，算法实现侧重点不同，具体细节比较麻烦；<br>但从整体上来看，都是先将 <strong>reads</strong> 间 <strong>overlap</strong> 关系构图，然后具体去简化这个图。<br>短 <strong>reads</strong> 一般采用基于 <strong>Kmer</strong> 的 De Brujin Graph(<strong>DBG</strong>)，<br>传统长 <strong>reads</strong> 一般采用 Overlap-Layout-Consensus（<strong>OLC*</strong>) 或 String Graph（<strong>SG</strong>)。<br>
    
  </div>
  <footer class="end-sep">
    
      
        <div class="alignleft">
          <a href="/2013/01/10/2013-01-10-assemble-a-genome/#more" class="more-link">Read More</a>
        </div>
      
      
    
    <div class="clearfix"></div>
  </footer>
</article>



  
    <article class="post">
  <header>
    
      <div class="icon"></div>
      <a href="/2013/01/04/2013-01-04-about-me/">
  <time datetime="2013-01-04T07:08:44.000Z">
    Jan 4 2013
  </time>
</a>
    
    
  
    <h1 class="title"><a href="/2013/01/04/2013-01-04-about-me/">About Me</a></h1>
  

  </header>
  
  <div class="entry">
    
      <h3 id="About-Me"><a href="#About-Me" class="headerlink" title="About Me !"></a>About Me !</h3><p>I’m a bioinformatic engineer and amateur programmer, and here is a short introduction about me and this blog, thanks for your reading!</p>
<p>I’m engaged in <strong>genomics research</strong>, including <strong><em>genome assembly/annotation/evolution</em></strong> and <strong><em>comparative genomics</em></strong>.</p>
<p>As a mathematics graduate, I’m also interested in interdisciplinary work, specifically <strong>data mining</strong> and <strong>visualization</strong>. And I hope we can exchange our ideas and experiences about  <strong>Data Mining</strong> through this blog.</p>
<p>I mostly work in <strong>python/perl/R</strong> on OSX and Linux for most of my work,  I also dabble in <strong>C/C++/Shell</strong>. And I am trying to learn <a href="http://d3js.org/" target="_blank" rel="noopener">D3.js</a> and <a href="http://www.processing.org/" target="_blank" rel="noopener">processing</a> for my hobby of <strong>Data Visualization</strong> now.</p>
<p>Thanks for following me: <a href="http://buttonwood.github.io">@Github</a> and <a href="http://weibo.com/buttonwood" target="_blank" rel="noopener">@Weibo</a>.</p>
<p>I’m living in Wuhan, a city near the Yangtze River in China. It’s beautiful besides it’s bad weather(long hot summer and cold winter), I spent my college life there and I love my friends there forever.</p>
<p>Besides programming, I exercise and read books regularly. To be a stronger and better version of myself!</p>
<h3 id="About-this-blog"><a href="#About-this-blog" class="headerlink" title="About this blog !"></a>About this blog !</h3><p>This blog is proudly powered by <a href="http://hexo.io/" target="_blank" rel="noopener">Hexo</a>, which takes great advantage of <a href="http://nodejs.org/" target="_blank" rel="noopener">Node.js</a>.</p>

<p>And it is only for my scattered ideas of reading and programming, there are no business attempts.</p>
<p>Any suggestions about this blog and topics talked there are welcome.</p>
<p>Like the TED said: <strong><em>Ideas are worth spreading and sharing!</em></strong> Many THX!</p>
<h3 id="Time-axis"><a href="#Time-axis" class="headerlink" title="Time axis !"></a>Time axis !</h3><div class="table-container">
<table>
<thead>
<tr>
<th>Time</th>
<th>Department/Works</th>
<th>Company/University</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong><em>2010-Now</em></strong></td>
<td>Bioinformatics Engineer/Analyst</td>
<td><strong><em><a href="http://www.genomics.cn/index" target="_blank" rel="noopener">BGI</a></em></strong></td>
</tr>
<tr>
<td><strong><em>2006-2010</em></strong></td>
<td>Student in College of Mathematics And Statistics</td>
<td><strong><em><a href="http://maths.hust.edu.cn/new/index.asp" target="_blank" rel="noopener">HUST</a></em></strong></td>
</tr>
</tbody>
</table>
</div>

    
  </div>
  <footer class="end-sep">
    
      
      
    
    <div class="clearfix"></div>
  </footer>
</article>



  
    <article class="post">
  <header>
    
      <div class="icon"></div>
      <a href="/2012/12/28/2012-12-28-about-this-blog/">
  <time datetime="2012-12-28T00:21:00.000Z">
    Dec 28 2012
  </time>
</a>
    
    
  
    <h1 class="title"><a href="/2012/12/28/2012-12-28-about-this-blog/">About This Blog!</a></h1>
  

  </header>
  
  <div class="entry">
    
      <h2 id="1-Bulletin-Board"><a href="#1-Bulletin-Board" class="headerlink" title="1.Bulletin Board"></a>1.Bulletin Board</h2><p>Hi, I’m <em><a href="http://weibo.com/buttonwood" target="_blank" rel="noopener">Buttonwood</a>!</em> This is my first blog on <a href="https://github.com" target="_blank" rel="noopener">github</a>.</p>
<p><em>Writing blogs like Hackers!</em></p>
<p>This is really very very cool!  And I’ve learned lots of things from it.</p>
<p>Yes, after days of struggling, my  <a href="http://buttonwood.github.com" target="_blank" rel="noopener"><strong><em>Techblog</em></strong></a> is online at  last!<br>Thx to the Internet and we‘re in an Open Society as well as  A Boom Time.</p>
<p>It is a personnal blog focus on new technology and thoughts, and also for  summing up the practical experience of my work  and life.    </p>
<p>It is nonbusiness and all right reserved.     </p>
<p>Besides, this is an individual experiment and my personal deed，it has nothing to do  with any company、organization or institution.     </p>
<p>Any suggestion about this blog is welcome, and I’ll keep on updating.</p>
    
  </div>
  <footer class="end-sep">
    
      
        <div class="alignleft">
          <a href="/2012/12/28/2012-12-28-about-this-blog/#more" class="more-link">Read More</a>
        </div>
      
      
    
    <div class="clearfix"></div>
  </footer>
</article>



  

  <nav id="pagination">
  
    <a href="/page/2/" class="prev">Prev</a>
  
  
  <div class="clearfix"></div>
</nav>

</div>
  </div>
  <footer id="footer"><div class="copyright">
  
  &copy; 2018 <a href="/">Buttonwood</a>
  
</div>
<div class="theme-copyright">
  Theme by <a href="https://github.com/orderedlist" target="_blank">orderedlist</a>
   | 
  Redesign by <a href="http://heroicyang.com/" target="_blank">Heroic Yang</a>
</div>
<div class="clearfix"></div></footer>
  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.8/jquery.min.js"></script>
<script src="/js/scale.fix.js"></script>
<script src="/js/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>


<script type="text/javascript">
  var duoshuoQuery = { short_name: 'buttonwood' };
  (function() {
    var ds = document.createElement('script');
    ds.type = 'text/javascript';
    ds.async = true;
    ds.src = 'http://static.duoshuo.com/embed.js';
    ds.charset = 'UTF-8';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(ds);
  })();
</script>



<link rel="stylesheet" href="/fancybox/jquery.fancybox.css" media="screen" type="text/css">
<script src="/fancybox/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
  (function($){
    $('.fancybox').fancybox();
  })(jQuery);
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        tex2jax: {
            inlineMath: [ ["$","$"], ["\\(","\\)"] ],
            skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code'],
            processEscapes: true
        }
    });
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax();
        for (var i = 0; i < all.length; ++i)
            all[i].SourceElement().parentNode.className += ' has-jax';
    });
</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<!--<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>--><!-- hexo-inject:begin --><!-- hexo-inject:end -->

</body>
</html>